flink-clients/src/main/java/org/apache/flink/client/program/ContextEnvironment.java

```diff
@@ -76,6 +76,7 @@ public JobExecutionResult execute(String jobName) throws Exception {
 			final JobExecutionResult  jobExecutionResult = getJobExecutionResult(jobClient);
 			jobListeners.forEach(jobListener ->
 					jobListener.onJobExecuted(jobExecutionResult, null));
+			this.lastJobExecutionResult = jobExecutionResult;
```

Comment:
Instead of using a local variable `jobExecutionResult` we could just always use the field `lastJobExecutionResult` here.

Reply:
I got you.